static/js/components/widgets/RelationField.tsx

```diff
@@ -155,20 +154,26 @@ export default function RelationField(
 
   const handleChange = useCallback(
     (event: any) => {
-      if (onChange) {
-        onChange(event)
-      } else if (setFieldValue) {
-        const content = event.target.value
-
-        // need to do this because we've renamed the
-        // nested field to get validation working
-        setFieldValue(name.split(".")[0], {
-          website: website.name,
-          content
-        })
+      // When we run renameNestedFields we add a '.content' suffix to the name of the field for RelationField
+      // because the data structure looks like this: { content, website }
+      // where 'content' is either a string or a list of strings.
+      // Validation by yup is easier to work with when the field name ends with .content because
+      // formik validates the value of 'content' instead of '{ content, website }'.
+      // But while that's easier for validation, we still need to send the whole object.
+      // So we need to remove that .content suffix and also add back 'website' when onChange is called.
+      const updatedEvent = {
+        target: {
+          name:  event.target.name.replace(/\.content$/, ""),
+          value: {
+            website: website.name,
+            content: event.target.value
+          }
+        }
       }
+      // @ts-ignore
```

Comment:
any way we can avoid the `@ts-ignore` here? I think generally it is a good goal for us to only use them in test files. Maybe you can add a cast on the type of `updatedEvent` or adjust the type signature for the `onChange` prop?

Reply:
Yes, it was not in the prop types actually so it should be easy to fix